server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -527,13 +526,10 @@ public void recoverFromTranslog(
         ActionListener<Void> listener
     ) {
         ActionListener.run(listener, l -> {
-            try (ReleasableLock lock = readLock.acquire()) {
-                ensureOpen();
-                try {
-                    translogRecoveryRunner.run(this, Translog.Snapshot.EMPTY);
-                } catch (final Exception e) {
-                    throw new EngineException(shardId, "failed to recover from empty translog snapshot", e);
-                }
+            try (var ignored = acquireEnsureOpenRef()) {
```

Comment:
I think this changes the exception handling (throwing an EngineException with an ACE as the cause)?

Reply:
Oh yes well-spotted indeed. Fixed in ecbedb55163601b46d5b120b4e0d4de183feeb6f.